<launch>

<arg name="num_joints" />

<arg name="controlled_index" />

<arg name="force_controller_hz" />

<arg name="force_controller_num_coeff" />

<arg name="force_controller_den_coeff" />

<arg name="force_controller_gain" default="1.0" />

<arg name="topic_measured_wrench" />

<arg name="ee_pose_topic" />

<arg name="change_wrench_frame" default="true" />

<node name="robot_as"  pkg="sun_robot_ros" type="robot_as" output="screen">

    <param name="num_joints" type="int" value="$(arg num_joints)"/>

    <!--
    <param name="hz" value="1000.0"/>
    -->

    <param name="service_get_clik_status" value="force_control/get_clik_status"/>

    <param name="service_set_clik_mode" value="force_control/set_clik_mode"/>

    <param name="topic_cartesian_command" value="force_control/desired_pose_twist_in"/>

</node>

<node name="wrench_ch_frame"  pkg="sun_utils" type="wrench_ch_frame" output="screen" if="$(arg change_wrench_frame)">

    <param name="use_rotation_only" type="bool" value="true"/>

    <param name="topic_wrench_in" value="$(arg topic_measured_wrench)"/>

    <param name="topic_wrench_out" value="force_control/wrench_in"/>

    <param name="topic_pose" value="$(arg ee_pose_topic)"/>

</node>

<node name="force_controller"  pkg="sun_robot_ros" type="force_controller" output="screen">

    <param name="topic_measured_wrench" value="force_control/wrench_in" if="$(arg change_wrench_frame)"/>
    <param name="topic_measured_wrench" value="$(arg topic_measured_wrench)" unless="$(arg change_wrench_frame)"/>

    <!--
    <param name="topic_desired_wrench" value="$(arg topic_desired_wrench)"/>
    -->

    <param name="controlled_index" value="$(arg controlled_index)"/>

    <param name="hz" value="$(arg force_controller_hz)"/>

    <rosparam param="num_coeff" subst_value="true">$(arg force_controller_num_coeff)</rosparam>

    <rosparam param="den_coeff" subst_value="true">$(arg force_controller_den_coeff)</rosparam>

    <param name="controller_gain" value="$(arg force_controller_gain)"/>

</node>

</launch>